src/Iterator/RandomIterator.php

```diff
@@ -79,9 +79,10 @@ public function key()
 
     public function next(): void
     {
-        unset($this->indexes[$this->key]);
-
-        $this->key = key($this->indexes);
+        if (null !== $this->key) {
+            unset($this->indexes[$this->key]);
+            $this->key = key($this->indexes);
+        }
```

Comment:
@drupol let me know if this looks right to you. If the key is _null_ then the `unset` will not have any effect anyway

Reply:
Nice catch :)

Reply:
Psalm's catch 😄